### PR TITLE
[compliance-agent] report resource_type and resource_id based on check's scope

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Report `resource_type` and `resource_id` based on check's scope

### Motivation

https://github.com/DataDog/architecture/pull/865

### CODE REVIEW

**Review by cswatt — COMMENTED** [review]
no docs review required :)

File: cmd/agent/install_script.sh [hunk 1]
```diff
@@ -343,7 +343,14 @@ elif [ "$OS" = "Debian" ]; then
...
     # see https://unix.stackexchange.com/q/146283 for reference - we use DEBIAN_FRONTEND=noninteractive to fix that
-    $sudo_cmd DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl gnupg
+    if [ -z "$sudo_cmd" ]; then
+        # if $sudo_cmd is empty, doing `$sudo_cmd X=Y command` fails with
+        # `X=Y: command not found`; therefore we don't prefix the command with
+        # $sudo_cmd at all in this case
+        DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl gnupg
+    else
+        $sudo_cmd DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl gnupg
+    fi
```
Comment (mx-psi) [inline cmd/agent/install_script.sh:353]: Merge/rebase gone wrong?

Comment (lebauce) [reply cmd/agent/install_script.sh]: Indeed :-) Fixed

File: generate_tools.go [hunk 2]
```diff
@@ -14,5 +14,6 @@ import (
 	_ "github.com/mailru/easyjson/easyjson"
 	_ "github.com/shuLhan/go-bindata/cmd/go-bindata"
 	_ "github.com/tinylib/msgp"
+	_ "github.com/vektra/mockery/v2"
```
Comment (mx-psi) [inline generate_tools.go:17]: Is this really a tool used in `go:generate`? If not it should go on `internal/tools/tools.go` and be installed via `inv install-tools`

Comment (lebauce) [reply generate_tools.go:17]: The tool is run using `go run github.com/vektra/mockery/v2` in a script (pkg/compliance/genso the idea was that it uses the version specified in go.mod
I don't know if the internal/tools/go.mod will be used in this case

Comment (lebauce) [reply generate_tools.go]: I modified internal/tools/go.mod and use the `mockery` binary instead of `go run`

Comment (mx-psi) [reply generate_tools.go]: Thanks! That way it doesn't end up in the licenses list :)

**Review by safchain — APPROVED** [review]

**Pull request merged**